src/components/Details.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCircleRight } from '@fortawesome/free-solid-svg-icons';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import TopCard from './TopCard';
+import Map4 from '../assets/images/map4.svg';
+
+const Details = () => {
+  const { location } = useParams();
+  const data = useSelector((state) => state.home.home);
+
+  const locationData = data.find((item) => item.location === location);
+  const [isSearchVisible, setIsSearchVisible] = useState(false);
+  const [searchKeyword, setSearchKeyword] = useState('');
+
+  useEffect(() => {
+    window.history.pushState({}, '', '/detail');
+    return () => {
+      window.history.pushState({}, '', '/');
+    };
+  }, []);
+
+  if (!locationData) {
+    return <p>No data available for this location.</p>;
+  }
+  // eslint-disable-next-line max-len
+  const filteredMeasurements = locationData.measurements.filter((metric) => metric.parameter.toLowerCase().includes(searchKeyword.toLowerCase()));
+  const toggleSearch = () => {
+    setIsSearchVisible(!isSearchVisible);
+    if (!isSearchVisible) {
+      setSearchKeyword('');
+    }
+  };
+  return (
+    <div>
+      <TopCard
+        backgroundImage={Map4}
+        location={locationData.location}
+        views="700 Views"
+        footerText="CITY/TOWN BREAKDOWN-2013"
+        setSearchKeyword={setSearchKeyword}
+        toggleSearch={toggleSearch}
+        isSearchVisible={isSearchVisible}
+      />
```

Comment:
- [ ] Kindly remove the unnecessary "700 views" text on the views on the details page, or provide more relevant information.

  ![image](https://github.com/RileyManda/air-quality/assets/36542986/137ff7c3-079f-4a8f-99fc-a2a766a880b3)